peer/brontide.go

```diff
@@ -2508,6 +2532,81 @@ func chooseDeliveryScript(upfront,
 	return upfront, nil
 }
 
+// restartCoopClose checks whether we need to restart the cooperative close
+// process for a given channel.
+func (p *Brontide) restartCoopClose(lnChan *lnwallet.LightningChannel) (
+	*lnwire.Shutdown, error) {
+
+	// If this channel has status ChanStatusCoopBroadcasted and does not
+	// have a closing transaction, then the cooperative close process was
+	// started but never finished. We'll re-create the chanCloser state
+	// machine and resend Shutdown. BOLT#2 requires that we retransmit
+	// Shutdown exactly, but doing so would mean persisting the RPC
+	// provided close script. Instead use the LocalUpfrontShutdownScript
+	// or generate a script.
```

Comment:
maybe just add to this comment "or generate a script if no previous one was set" or something? cause the comment seems contradictory

Reply:
there's no contradiction, there's a "but" to indicate we don't do this

Reply:
but then if there is no delivery script then we do generate a new one (L2562)? probably me just misunderstanding. Just want to double check what is meant here

Reply:
you're not supposed to generate a new delivery script according to the spec, it would have to be persisted and re-sent on startup

Reply:
I think this can be fixed in a later PR. We currently don't do shutdown correctly (and neither does anybody else), but this PR in isolation is a win for lnd

Reply:
> 

All we do here is fetch a delivery script, generating one if we didn't set an upfront

Reply:
gotcha 👍